tests/ballerina-unit-test/src/test/resources/test-src/services/configuration/compression-annotation-test.bal

```diff
@@ -56,3 +95,29 @@ service<http:Service> userOverridenValue bind mockEP {
         _ = conn -> respond(res);
     }
 }
+
+@http:ServiceConfig {compression: {contentTypes:["text/plain"]}}
+service<http:Service> autoCompressWithInCompatibleContentType bind mockEP {
+    @http:ResourceConfig {
+        methods:["GET"],
+        path:"/"
+    }
+    test1 (endpoint conn, http:Request req) {
+        http:Response res = new;
+        res.setJsonPayload({ test: "testValue" }, contentType = "application/json;charset=\"ISO_8859-1:1987\"");
+        _ = conn -> respond(res);
+    }
+}
+
+@http:ServiceConfig {compression: {enable: http:COMPRESSION_ALWAYS, contentTypes:[]}}
+service<http:Service> alwaysCompressWithEmptyContentType bind mockEP {
+    @http:ResourceConfig {
+        methods:["GET"],
+        path:"/"
+    }
+    test1 (endpoint conn, http:Request req) {
+        http:Response res = new;
+        res.setTextPayload("Hello World!!!");
+        _ = conn -> respond(res);
+    }
+}
```

Comment:
Since we take an array of strings as a list of content types, we don't' know whether the user species a  content type in the correct format. Can you please add a negative test case for this.

Reply:
Added via aedb63f.